_transcripts/006-rust-1.39.0.md

```diff
@@ -0,0 +1,783 @@
+that kind of choice on our consumers. Rust is a very low level language, and
+that means that we are cursed with having to be modular and supporting many,
+many use cases. And so no matter what we shipped, somebody would have a problem
+with it. And so at the end of the day, we have to let you swap in and out
+different ways of running your green threads.
+
+__Jon__: Yeah. One way to think about why this is necessary is, imagine that
+you're running on, like, an embedded device that maybe doesn't even have an
+operating system. You can't use sort of a normal executor, the same way you
+would, like, on a Windows or Linux machine, right? Because you might not have
+access to system calls. You might not have the same ability to do things like
+epoll or use IOCP or any of these features that the executor might rely on. And
+so you might want to write a custom executor specifically for that environment.
+And so if Rust just shipped with an executor that everyone had to use, you
+wouldn't have that option.
+
+__Ben__: Yeah, I believe there is actually— there is a blog post that was
+written on the official Rust blog about when the prior release came out, kind of
+just talking about various things, including the ecosystem of things. And so it
+was talking about, like, I believe Tokio is planning a major release soon. The
+`futures-rs` library, which is kind of the playground for things that might
+someday end up in the standard library, just had a 0.3 release. `async-std` as
+of about a week or so ago, just released their 1.0. So that's a great thing.
+I've seen plenty of other libraries start to support `async-std`, there's a
+really good developing ecosystem there.
+
+__Jon__: I thought one thing that was cool from that blog post, which we'll link
+in the show notes, was this `wasm_bindgen_futures` crate. So this is a crate
+that lets you turn Rust— it lets you bridge between Rust futures and JavaScript
+promises. So if you're using wasm, it lets you— I don't know exactly how the
+binding works, but it lets you wait on JavaScript promises in Rust and let you
+wait on Rust futures in JavaScript.
+
+__Ben__: Very cool.
+
+__Jon__: Which is just really cool. The fact that this even possible is amazing.
+
+__Ben__: What else is there talk about, do you think? We talked about async
+closures. Talked about lots of this stuff. Is that kind of it? It wasn't nearly
+as long as we thought it might take.
+
+__Jon__: That's true. I think one thing that I want to mention about
+async/await. And I think one reason why people are so excited, as an explanation
+for those who haven't used async/await, and don't know why they should be
+excited. One thing that's really neat about async/await is how uninteresting it
+ends up making your code. Because with the old future stuff you had to, like, do
+this— this is known as, sort of, callback hell in JavaScript, for example, where
+you have to write code like `future1.and_then(future2.and_then(...))`
```

Comment:
I'm pretty sure this isn't valid javascript, but I'm not sure how to bridge the gap between what was said and what would actually be valid syntax.

Reply:
Hmm, not sure. This may be good enough I think. Alternatively maybe:
`future1.and_then(function(v1) { return future2.and_then(function(v2) { ... }); }`

Reply:
I'll go along with whatever you prefer.  I won't usually do aggressive corrections like this on my own-- it's not my place as the transcriber to edit others' words.

Reply:
How about:
`future1.and_then` ... `future2.and_then`
?

Reply:
Looks good, pushed a follow-up commit with that.